### PR TITLE
Add github workflow

### PR DESCRIPTION
**Pull request by lagoan**
## Context

Adding github workflow brings the project in line with Jupiter following the same testing mechanism.

## What's New

This PR includes files describing the github workflow for CI, a template for PR descriptions, and definition of code owners.

### CODE REVIEW

File: .github/workflows/ruby.yml [hunk 1]
```diff
@@ -0,0 +1,32 @@
...
+  pull_request:
+    branches: [ master ]
+
+jobs:
+  test:
+    runs-on: ubuntu-latest
+    steps:
+    - uses: actions/checkout@v2
+    - name: Build the docker-compose stack
+      run: docker-compose -f docker-compose-ci.yml up -d
```
Comment (murny) [inline .github/workflows/ruby.yml:22]: if all we need is redis, then lets use github actions service? No need for this docker stuff. 

The reason you see this in Jupiter is we have a very weird dependency with Solr. We need our code checkout, which we push config into solr, but if we use services, we don't have access to our code. Basically chicken and egg problem.

Here you can just use this:

```
jobs:
  test:
    runs-on: ubuntu-latest

    services:
      redis:
        image: redis:alpine
        ports: ["6379:6379"]
 steps:
    - uses: actions/checkout@v2
 ....etc...
```

File: .github/workflows/ruby.yml [hunk 2]
```diff
@@ -0,0 +1,32 @@
+# This workflow uses actions that are not certified by GitHub.
+# They are provided by a third-party and are governed by
+# separate terms of service, privacy policy, and support
+# documentation.
+# This workflow will download a prebuilt Ruby version, install dependencies and run tests with Rake
+# For more information see: https://github.com/marketplace/actions/setup-ruby-jruby-and-truffleruby
```
Comment (murny) [inline .github/workflows/ruby.yml:6]: Do we need this disclaimer?

File: .github/workflows/ruby.yml (hunk 1 above)
Comment (lagoan) [reply .github/workflows/ruby.yml]: Thanks! This is definitely and improvement.

File: .github/workflows/ruby.yml (hunk 2 above)
Comment (lagoan) [reply .github/workflows/ruby.yml]: No we dont

File: .github/workflows/ruby.yml [hunk 3]
```diff
@@ -0,0 +1,27 @@
...
+    - name: Set up Ruby
+      uses: ruby/setup-ruby@v1
+      with:
+        ruby-version: 2.6.6
+    - name: Install dependencies
+      run: bundle install
+    - name: Lint with RuboCop
+      run: bundle exec rubocop --parallel
+    - name: Run tests
+      run: bundle exec rake spec
```
Comment (murny) [inline .github/workflows/ruby.yml:27]: Should we add Danger?

Comment (lagoan) [reply .github/workflows/ruby.yml:31]: We should indeed!

Comment (lagoan) [reply .github/workflows/ruby.yml:31]: You are right about the travisCI stuff. Will be running the checks in parallel for a bit, after we make sure everything is OK will be sending another PR removing travisCI.

**Pull request merged**